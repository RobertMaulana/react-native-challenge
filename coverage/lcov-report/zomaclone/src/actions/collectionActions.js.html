<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for zomaclone/src/actions/collectionActions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">zomaclone/src/actions</a> collectionActions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { GET_COLLECTIONS } from './constants'
import axios from 'axios'
&nbsp;
export const getCollections = collections =&gt; ({
  type: GET_COLLECTIONS,
  payload: collections
})
&nbsp;
export const fetchCollections = () =&gt; (
  dispatch =&gt;
  // axios.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&amp;count=50',
  // {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  // .then(function (response) {
  //   dispatch(getCollections(response.data.collections))
  // })
  // .catch(function (err) {
  //   console.log(err.message);
  // })
  fetch('https://developers.zomato.com/api/v2.1/collections?city_id=1&amp;count=50',
  {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  .then(res =&gt; res.json())
  .then(data =&gt; dispatch(getCollections(data.collections)))
  .catch(<span class="fstat-no" title="function not covered" >fu</span>nction (err) {
<span class="cstat-no" title="statement not covered" >    console.log(err.message);</span>
  })
)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 04 2017 15:09:03 GMT+0700 (WIB)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
