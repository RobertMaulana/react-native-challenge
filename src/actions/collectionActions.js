import { GET_COLLECTIONS } from './constants'
import axios from 'axios'

export const getCollections = collections => ({
  type: GET_COLLECTIONS,
  payload: collections
})

export const fetchCollections = () => (
  dispatch =>
  // axios.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50',
  // {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  // .then(function (response) {
  //   dispatch(getCollections(response.data.collections))
  // })
  // .catch(function (err) {
  //   console.log(err.message);
  // })
  fetch('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50',
  {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  .then(res => res.json())
  .then(data => dispatch(getCollections(data.collections)))
  .catch(function (err) {
    console.log(err.message);
  })
)
