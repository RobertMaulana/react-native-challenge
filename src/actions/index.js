import { fetchCollections } from './collectionActions'

export {
  fetchCollections
}
