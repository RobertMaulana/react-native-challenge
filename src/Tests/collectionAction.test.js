import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import fetchMock from 'fetch-mock'
// import moxios from 'moxios'

import { GET_COLLECTIONS } from '../actions/constants'
import { fetchCollections, getCollections } from '../actions/collectionActions'

const mockStore = configureMockStore([thunk])

describe('Collection action', () => {

  it('should run get action', () => {
    expect(getCollections(1000)).toEqual(
      {
        type: GET_COLLECTIONS,
        payload: 1000
      }
    )
    expect(getCollections(1000)).toMatchSnapshot()
  })

  it('should run fetch action', () => {
    const store = mockStore()
    fetchMock.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50', { collections: ["test"] })
    return store.dispatch(fetchCollections())
    .then(()=> {
      expect(store.getActions()).toEqual([
        {
          payload: ["test"],
          type: GET_COLLECTIONS
        }
      ])

    })
  })
})
