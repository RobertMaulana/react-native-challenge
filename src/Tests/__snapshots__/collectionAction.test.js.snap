// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection action should run get action 1`] = `
Object {
  "payload": 1000,
  "type": "GET_COLLECTIONS",
}
`;
